Src/Test.UnitTests.Sarif.PatternMatcher.Cli/AnalyzeCommandTests.cs

```diff
@@ -89,12 +92,66 @@ public void AnalyzeCommand_IntrafileBasic()
+
+            string definitionsText = GetIntrafileRuleDefinition();
+
+            string fileContents = "unused leading space  \r\n" +
+                                  " secret1              \r\n" +
+                                  " host1                \r\n" +
+                                  " id1                  \r\n" +
+                                  " secret2              \r\n" +
+                                  "unused trailing space \r\n";
+
+            SarifLog sarifLogWithLargeFileExcluded = RunAnalyzeCommandWithFileSizeInKilobytes(
+                definitionsText,
+                fileContents,
+                fileSizeInKilobytes: 1);
+
+            SarifLog sarifLogWithLargeFileIncluded = RunAnalyzeCommandWithFileSizeInKilobytes(
+                definitionsText,
+                fileContents,
+                fileSizeInKilobytes: 1024);
+
+            SarifLog sarifLogOnlySmallFile = RunAnalyzeCommand(
+                definitionsText,
+                fileContents);
+
+            sarifLogWithLargeFileExcluded.Should().NotBeNull();
+            sarifLogWithLargeFileExcluded.Runs?[0].Results?.Count.Should().Be(2);
+            sarifLogWithLargeFileIncluded.Should().NotBeNull();
+            sarifLogWithLargeFileIncluded.Runs?[0].Results?.Count.Should().Be(4);
+            sarifLogWithLargeFileIncluded.Runs[0].Artifacts.Count.Should().Be(2);
+
+            Assert.True(string.Equals(sarifLogWithLargeFileIncluded.Runs[0].Artifacts[0].Location.Uri, smallTargetPath) ||
+                string.Equals(sarifLogWithLargeFileIncluded.Runs[0].Artifacts[0].Location.Uri, largeTargetPath));
+
+            Assert.True(string.Equals(sarifLogWithLargeFileIncluded.Runs[0].Artifacts[1].Location.Uri, smallTargetPath) ||
+                string.Equals(sarifLogWithLargeFileIncluded.Runs[0].Artifacts[1].Location.Uri, largeTargetPath));
+
+            sarifLogWithLargeFileExcluded.Runs[0].Results[0].Locations.Should()
+                .BeEquivalentTo(sarifLogOnlySmallFile.Runs[0].Results[0].Locations);
+
+            sarifLogWithLargeFileExcluded.Runs[0].Results[1].Locations.Should()
+                .BeEquivalentTo(sarifLogOnlySmallFile.Runs[0].Results[1].Locations);
+
+            sarifLogWithLargeFileExcluded.Runs[0].Artifacts.Count.Should().Be(1);
+
+            Assert.True(string.Equals(sarifLogWithLargeFileExcluded.Runs[0].Artifacts[0].Location.Uri, smallTargetPath) &&
+                !string.Equals(sarifLogWithLargeFileExcluded.Runs[0].Artifacts[0].Location.Uri, largeTargetPath));
+        }
+
         private SarifLog RunAnalyzeCommand(string definitionsText, string fileContents)
         {
             string sarifOutput;
-            string rootDirectory = @"e:\repros";
-            string scanTargetName = $"test.txt";
-            string scanTargetPath = @$"{rootDirectory}\{scanTargetName}";
+            string rootDirectory = Directory.GetCurrentDirectory();
+            string scanTargetName = SmallTargetName;
```

Comment:
>scanTargetName [](http://example.com/codeflow?start=19&length=14)

don't you need to make this accept the targetName? small and large?

Reply:
No, this is to preserve the original functionality of this test, which relied on mocks and will throw an unhandled exception without a valid file to check the size of.